internal/services/compute/virtual_machine_scale_set_data_source.go

```diff
@@ -164,6 +164,9 @@ func dataSourceVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta interf
 		if instance.InstanceID != nil {
 			nics, err := networkInterfacesClient.ListVirtualMachineScaleSetVMNetworkInterfacesComplete(ctx, id.ResourceGroup, id.Name, *instance.InstanceID)
 			if err != nil {
+				if utils.ResponseWasNotFound(nics.Response().Response) {
+					break
```

Comment:
`break` is going to cancel retrieving all of the NICs, I think this instead wants to become:

```suggestion
					if err := result.NextWithContext(ctx); err != nil {
						return fmt.Errorf("listing next page VM Instances for Virtual Machine Scale Set %q (Resource Group %q): %+v", id.ResourceGroup, id.Name, err)
					}
					continue
```